File: ObjectAvoidance/java/src/main/java/elite/math/angle/ArcMinute.java
package elite.math.angle;

/**
 *
 * @author CFerg (Elite)
 */
public class ArcMinute{
    
    double minute;
    
    /**
     *
     */
    public ArcMinute(){
        minute = (1/60);
    }
    
    /**
     *
     * @param a
     */
    public ArcMinute(Degree a){
        this.minute = (a.degree/60);
    }
    
    /**
     *
     * @param a
     */
    public ArcMinute(ArcSecond a){
        this.minute = (a.second*60);
    }
    
    /**
     *
     * @param a
     */
    public ArcMinute(MilliArcSecond a){
        this.minute = ((a.milliSecond*60)*1000);
    }
    
    /**
     *
     * @param a
     */
    public ArcMinute(MicroArcSecond a){
        this.minute = (((a.microSecond*60)*1000)*1000);
    }
    
    /**
     *
     * @return
     */
    public Degree toDegree(){
        return new Degree(this);
    }
    
    /**
     *
     * @return
     */
    public ArcSecond toSecond(){
        return new ArcSecond(this);
    }
    
    /**
     *
     * @return
     */
    public MilliArcSecond toMilliSecond(){
        return new MilliArcSecond(this);
    }
    
    /**
     *
     * @return
     */
    public MicroArcSecond toMicroSecond(){
        return new MicroArcSecond(this);
    }
}

File: ObjectAvoidance/src/elite/math/angle/ArcMinute.kt
package elite.math.angle

/**
 *
 * @author CFerg (Elite)
 */
class ArcMinute {

    internal var minute: Double = 0.toDouble()

    /**
     *
     */
    constructor() {
        minute = (1 / 60).toDouble()
    }

    /**
     *
     * @param a
     */
    constructor(a: Degree) {
        this.minute = a.degree / 60
    }

    /**
     *
     * @param a
     */
    constructor(a: ArcSecond) {
        this.minute = a.second * 60
    }

    /**
     *
     * @param a
     */
    constructor(a: MilliArcSecond) {
        this.minute = a.milliSecond * 60 * 1000
    }

    /**
     *
     * @param a
     */
    constructor(a: MicroArcSecond) {
        this.minute = a.microSecond * 60 * 1000 * 1000
    }

    /**
     *
     * @return
     */
    fun toDegree(): Degree {
        return Degree(this)
    }

    /**
     *
     * @return
     */
    fun toSecond(): ArcSecond {
        return ArcSecond(this)
    }

    /**
     *
     * @return
     */
    fun toMilliSecond(): MilliArcSecond {
        return MilliArcSecond(this)
    }

    /**
     *
     * @return
     */
    fun toMicroSecond(): MicroArcSecond {
        return MicroArcSecond(this)
    }
}